### PR TITLE
SAAS-20313/Front-Door-Request-Body-Inspection

### PR DESCRIPTION
**Pull request by alphadev4**
Please merge this pr first since it has the api call added in api.js and location.js https://github.com/aquasecurity/cloudsploit/pull/1737

### CODE REVIEW

File: plugins/azure/frontdoor/frontDoorRequestBodyInspection.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door Request Body Inspection',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    description: 'Ensures that request body inspection is enabled for Azure Front Door premium WAF policy.',
+    more_info: 'Web Application Firewalls associated to Azure Front Doors premium that have request body inspection enabled, allows to inspect properties within the HTTP body that may not be evaluated in the HTTP headers, cookies, or URI.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorRequestBodyInspection.js:9]:
```suggestion
    more_info: 'Web Application Firewalls associated to Azure Front Doors that have request body inspection enabled, allows to inspect properties within the HTTP body that may not be evaluated in the HTTP headers, cookies, or URI.',
```

File: plugins/azure/frontdoor/frontDoorRequestBodyInspection.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    recommended_action: 'Ensure that request body inspection in policy settings for Azure Front Door WAF policy is enabled.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorRequestBodyInspection.js:10]:
```suggestion
    recommended_action: ''Modify Front Door WAF policy and enable request body inspection in policy settings.',
```

File: plugins/azure/frontdoor/frontDoorRequestBodyInspection.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            if (afdWafPolicies.err || !afdWafPolicies.data) {
+                helpers.addResult(results, 3, 'Unable to query for Front Door WAF policies: ' + helpers.addError(afdWafPolicies), location);
+                return rcb();
+            }
+            if (!afdWafPolicies.data.length) {
+                helpers.addResult(results, 0, 'No existing Front Door WAF policies found', location);
+                return rcb();
+            }
+
+            var frontDoorWafPolicies = false;
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorRequestBodyInspection.js:35]:
```suggestion
```

File: plugins/azure/frontdoor/frontDoorRequestBodyInspection.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            for (let policy of afdWafPolicies.data) {
+                if (!policy.id || !policy.sku || policy.sku.name.toLowerCase() != 'premium_azurefrontdoor') continue;
+
+                frontDoorWafPolicies = true;
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorRequestBodyInspection.js:39]:
```suggestion
```

File: plugins/azure/frontdoor/frontDoorRequestBodyInspection.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                    helpers.addResult(results, 0, 'Front Door WAF policy has request body inspection enabled', location, policy.id);
+                } else {
+                    helpers.addResult(results, 2, 'Front Door WAF policy does not have request body inspection enabled', location, policy.id);
+                }
+            }
+
+            if (!frontDoorWafPolicies) {
+                helpers.addResult(results, 0, 'No existing Front Door WAF policies found', location);
+            }
+
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorRequestBodyInspection.js:52]:
```suggestion

```

File: plugins/azure/frontdoor/frontDoorRequestBodyInspection.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door Request Body Inspection',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    description: 'Ensures that request body inspection is enabled for Azure Front Door WAF policy.',
+    more_info: 'Web Application Firewalls associated to Azure Front Doors that have request body inspection enabled, allows to inspect properties within the HTTP body that may not be evaluated in the HTTP headers, cookies, or URI.',
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/frontDoorRequestBodyInspection.js:9]:
```suggestion
    more_info: 'Web Application Firewalls associated to Azure Front Doors that have request body inspection enabled allow to inspect properties within the HTTP body that may not be evaluated in the HTTP headers, cookies, or URI.',
```

**Comment by mehakseedat63** [comment]
Fix failing test

**Pull request merged**